### PR TITLE
Allow omitting some optional fields

### PR DESCRIPTION
**Pull request by auscompgeek**
Presence, membership, and profile changes have optional fields.
These should be omitted rather than null when unwanted.

Fixes: #79

### CODE REVIEW

**Comment by turt2live** [comment]
Many of these are null for specific reasons, largely for server compatibility in off-spec behaviour.

**Comment by auscompgeek** [comment]
Hmm. Do you have any documentation for these broken behaviours anywhere?

**Comment by turt2live** [comment]
nope, hence "off-spec" :p 

Most of it is synapse, though there's other implementations out there that rely on a literal null value.

**Comment by auscompgeek** [comment]
Hmmm. Well, we already have clients that omit the kick/ban reason when unspecified, and we know #79 is a problem with Synapse. If it's just the profile change endpoints that have problems with being omitted, I can change those signatures back I guess.

**Comment by turt2live** [comment]
tbh I'd rather push forward on the spec side than trya nd fix/document all the problems in the library.

File: src/MatrixClient.ts [hunk 1]
```diff
@@ -269,11 +269,11 @@ export class MatrixClient extends EventEmitter {
     /**
      * Sets the presence status for the current user.
      * @param {"online"|"offline"|"unavailable"} presence The new presence state for the user.
-     * @param {string} statusMessage Optional status message to include with the presence.
+     * @param {string?} statusMessage Optional status message to include with the presence.
      * @returns {Promise<any>} Resolves when complete.
      */
     @timedMatrixClientFunctionCall()
-    public async setPresenceStatus(presence: "online" | "offline" | "unavailable", statusMessage: string = null): Promise<any> {
+    public async setPresenceStatus(presence: "online" | "offline" | "unavailable", statusMessage: string = undefined): Promise<any> {
```
Comment (Sorunome) [inline src/MatrixClient.ts:276]: also nicer syntax is `statusMessage?: string`, same applies to all the other things

**Comment by turt2live** [comment]
closing this per spec concerns

**Pull request closed without merging**